js/index.js

```diff
@@ -1,30 +1,42 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+  const price = product.querySelector(".price span") 
+  // --> Is this the best approach to get single strings, numbers, etc (not arrays)?
+  const priceNum= parseInt(price.innerText)
 
-  //... your code goes here
+  const quantity = product.querySelector(".quantity input")
+  const quantityNum= parseInt(quantity.value)
+
+  const subtotal = product.querySelector(".subtotal")
+  subtotal.innerText = priceNum * quantityNum
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
+  
   // ITERATION 2
-  //... your code goes here
 
+  const all = document.getElementsByClassName("product") 
+  //--> if I want to get an array, is it better to use getElementsByID?
+  const allArr = [...all];
+  allArr.forEach((product) => updateSubtotal(product));
+    
   // ITERATION 3
-  //... your code goes here
+
+  const subtotal = document.getElementsByClassName("subtotal");
+  const subtotalArr = [...subtotal];
+  const total = subtotalArr.reduce((accumulator, currentValue) => Number(accumulator.innerText) + Number(currentValue.innerText))
+  const totalValue = document.querySelector ("#total-value span")
+  totalValue.innerText = total;
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
+  target.parentNode.parentNode.parentNode.removeChild(target.parentNode.parentNode);
+  // console.log (productRem);
```

Comment:
when you remove 1 product, you need to update total value ;)

Reply:
Damn! so close....